### PR TITLE
ref(trace): cleanup state

### PR DESCRIPTION
**Pull request by JonasBa**
Centralizing the mess we used to call state and grouping components in a more logical way. This also gives us better control flow overview which was previously very hard to follow due to the number of useEffects everywhere...

Majority of the code had just been moved and renamed as we had some confusing and duplicate namings. I also got rid of most of the react effects that control flow hard to follow.

### CODE REVIEW

**Comment by JonasBa** [comment]
@Abdkhan14 we'll review this in a 1:1 meeting, I'll explain the reorg and test changes

**Review by Abdkhan14 — APPROVED** [review]
We paired and went over all the changes.
